### PR TITLE
New Materialize CSS Cheat Sheet

### PR DESCRIPTION
**Pull request by midhruvjaink**
New Materialize CSS Cheat Sheet

---

https://duck.co/ia/view/materialize

### CODE REVIEW

**Comment by daxtheduck** [comment]
@midhruvjaink Thanks for the pull request! Unfortunately, we're only accepting contributions and improvements to a limited portion of the search space.
            
The staff and community are currently focused on making DuckDuckGo the [best search engine for programmers](https://forum.duckduckhack.com/t/duckduckhack-programming-mission-overview/53). To ensure we're making as much progress as possible, **we are only accepting Instant Answers and updates related to the programming mission**.
    
If this is not related to the Programming Mission, it will likely be closed but you can:
    
1. Help us prevent further confusion by telling us where you found out about DuckDuckHack or where we could've alerted you earlier.
2. Keep this code handy! We'll be expanding into other search topics in the future!
3. Join us at our new home: https://forum.duckduckhack.com/

If this is related to the Programming Mission, it will be reviewed by staff and the community as soon as possible!

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: share/goodie/cheat_sheets/json/materialize.json [hunk 1]
```diff
@@ -0,0 +1,174 @@
+{
+    "id":"materialize-cheatsheet",
```
Comment (pjhampton) [inline share/goodie/cheat_sheets/json/materialize.json]: remove `-cheatsheet` from the id :)

File: share/goodie/cheat_sheets/json/materialize.json [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+    "name":"Materialize CSS",
+    "description":"Materialize CSS classes and cheat sheet",
+    "metadata":{
+        "sourceName":"Materialize CSS",
+        "sourceUrl":"http://materializecss.com/"
+    },
+    "aliases":[
+        "Materialize", "Material", "CSS", "CSS Frameworks"
```
Comment (pjhampton) [inline share/goodie/cheat_sheets/json/materialize.json]: Please make sure all of these are lowercase. This is one of the reasons why the tests are failing.

File: share/goodie/cheat_sheets/json/materialize.json [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+        "Badge":[
+            {
+                "val":"Collection",
+                "key":".collection"
+            },{
+                "val":"Badge",
+                "key":".badge"
+            },{
+                "val":"New badge",
+                "key":".new badge"
```
Comment (pjhampton) [inline share/goodie/cheat_sheets/json/materialize.json]: Is `badge` supposed to be `.badge`?

File: share/goodie/cheat_sheets/json/materialize.json [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+            },{
+                "val":"Badges in Dropdown",
+                "key":".dropdown-content"
+            },{
+                "val":"Color Badge",
+                "key":".new badge color"
```
Comment (pjhampton) [inline share/goodie/cheat_sheets/json/materialize.json]: Same with `.badge` and `.color`?

File: share/goodie/cheat_sheets/json/materialize.json [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+            }
+        ],
+        "Buttons":[
+            {
+                "val":"Waves effect, raised button",
+                "key":".waves-effect waves-light btn"
```
Comment (pjhampton) [inline share/goodie/cheat_sheets/json/materialize.json]: Same here?

File: share/goodie/cheat_sheets/json/materialize.json [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+                "val":"Card title",
+                "key":".card-title"
+            },{
+                "val":"Card action and links",
+                "key":".card-action"
+            },{
+                "val":"Card image",
+                "key":".card-image"
+            },{
+                "val":"Card reveal- reveals some more info on click",
```
Comment (pjhampton) [inline share/goodie/cheat_sheets/json/materialize.json]: This is the fist time you have had `name` - `explanation`. Maybe we should remove the > reveals some more info on click",

File: share/goodie/cheat_sheets/json/materialize.json [hunk 7]
```diff
@@ -0,0 +1,174 @@
+{
+    "id":"materialize-cheatsheet",
+    "name":"Materialize CSS",
+    "description":"Materialize CSS classes and cheat sheet",
```
Comment (pjhampton) [inline share/goodie/cheat_sheets/json/materialize.json]: Hmm, consider changing this to `Materialize CSS Classes`. That might be nicer :-)

**Comment by pjhampton** [comment]
Hey @midhruvjaink!!

This is awesome 😃  Thanks so much!! Don't forget to head over to the forum and help us with DuckDuckGo's programming mission: https://forum.duckduckhack.com/

In future, it's always a good idea to create an IA page at: duck.co

Thanks again!

**Comment by moollaza** [comment]
@midhruvjaink thanks a lot for making this!

Before we can deploy this to Beta.duckduckgo.com you need to create an IA Page and put the link into your PR description at the top of this page.

Once that has been done we can move along this PR 👍

**Comment by tagawa** [comment]
Hi @midhruvjaink. You can create a new IA (Instant Answer) page here: https://duck.co/ia/new_ia

Just fill in the boxes and when the page is made, paste its URL into the description at the top of this page. And of course, please let us know if you have any problems or questions!

File: share/goodie/cheat_sheets/json/materialize.json (hunk 1 above)
Comment (jophab) [inline share/goodie/cheat_sheets/json/materialize.json]: Change id to "materialize_cheat_sheet"

**Comment by midhruvjaink** [comment]
hey @tagawa @moollaza I have created the instant answer page - https://duck.co/ia/view/materialize
now what to do next

File: share/goodie/cheat_sheets/json/materialize.json [hunk 8]
```diff
@@ -0,0 +1,174 @@
+{
+    "id":"materialize_cheat_sheet",
+    "name":"Materialize CSS",
+    "description":"Materialize CSS Classes",
+    "metadata":{
+        "sourceName":"materialize css",
+        "sourceUrl":"http://materializecss.com/"
+    },
+    "aliases":[
+        "Materialize", "material", "css", "css frameworks"
```
Comment (tagawa) [inline share/goodie/cheat_sheets/json/materialize.json]: Please remove "Materialize" - it's the name of the cheat sheet so not needed here.

Comment (tagawa) [inline share/goodie/cheat_sheets/json/materialize.json]: Please remove "Materialize" - it's the name of the cheat sheet so not needed here.

File: share/goodie/cheat_sheets/json/materialize.json [hunk 9]
```diff
@@ -0,0 +1,174 @@
...
+    "sections":{
+        "Grid System":[
+            {
+                "val":"Sets pre-defined width for the content",
+                "key":".container"
+            },{
+                "val":"Columns",
+                "key":".col s1-s12"
+            },{
+                "val":"Offsets ",
```
Comment (tagawa) [inline share/goodie/cheat_sheets/json/materialize.json]: Please remove the space after `Offsets`

**Comment by tagawa** [comment]
@midhruvjaink Thank you for making the IA page. I've put the link in the description for our automation process. There are a couple of changes to make, which I've commented in the code, and when that's done the cheat sheet should be put on our beta server automatically for testing.

**Comment by midhruvjaink** [comment]
someone please recheck this again

**Comment by pjhampton** [comment]
Hey @midhruvjaink. Can you please address my questions above?

File: share/goodie/cheat_sheets/json/materialize.json [hunk 10]
```diff
@@ -0,0 +1,174 @@
+{
+    "id":"materialize_cheat_sheet",
+    "name":"Materialize CSS",
+    "description":"Materialize CSS Classes",
+    "metadata":{
+        "sourceName":"materialize css",
+        "sourceUrl":"http://materializecss.com/"
+    },
+    "aliases":[
+        "material", "css", "css frameworks"
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/materialize.json]: `css` and `css frameworks` are too broad, only aliases related to the Materialize framework should be used.

If someone searches for "css cheatsheet", this should not be the answer shown. The same goes for "css frameworks"

**Comment by pjhampton** [comment]
Closing as this PR looks to have gone stale.

If you are looking for something more substantial to work through that will have a big impact for the users of DuckDuckGo, please check out the [DuckDuckHack Issues Table](https://duckduckhack.com/issues). 🚀  😄

**Pull request closed without merging**